### PR TITLE
Add EX to set of the cart integration

### PR DESCRIPTION
**Pull request by Isengo1989**
### 1. Why is this change necessary?
Currently there is no expiration on the redis cart. The cleanup scheduled task only handles the mysql integration, therefor a solution for the redis cart is needed (otherwise the redis cart DB will build up and take a lot of memory). Since allkeys-lru is not supported anymore by the ShopwareRedisAdapter and there are many shops that do not have multiple redis instances this will make sense.

### 2. What does this change do, exactly?
It adds an Expiration in seconds to the key, so that it can be invalidated/deleted after a certain amount of time or when memory is needed.

### 3. Describe each step to reproduce the issue or behaviour.
Run the cart with redis and wait 2-3 weeks on a high load enviroment, the expire_days are ignored and all carts will be kept which leads to many GB of redis carts.

### 4. Please link to the relevant issues (if any).
none

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/workflow/2020-08-03-implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Checkout/Cart/Command/CartMigrateCommand.php [hunk 1]
```diff
@@ -231,9 +235,9 @@ private function sqlToRedis(InputInterface $input, OutputInterface $output): int
                     'line_item_count' => $row['line_item_count'],
                     'created_at' => $row['created_at'],
                 ]);
-            }
 
-            $this->redis->mset($values);
+                $this->redis->set($key, $value, ['EX' => $this->expireDays * 86400]);
```
Comment (shyim) [inline src/Core/Checkout/Cart/Command/CartMigrateCommand.php:239]: maybe redis pipeline here?

Comment (Isengo1989) [reply src/Core/Checkout/Cart/Command/CartMigrateCommand.php:239]: @shyim apperently predis and phpredis don't support pipelining with Redis Clusters. Will the PR be rejected if i don't use it?

File: src/Core/Checkout/DependencyInjection/cart.xml [hunk 2]
```diff
@@ -393,6 +393,7 @@
             <argument type="service" id="event_dispatcher"/>
             <argument type="service" id="Shopware\Core\Checkout\Cart\CartSerializationCleaner"/>
             <argument>%shopware.cart.compress%</argument>
+            <argument>%shopware.cart.expire_days%%</argument>
```
Comment (keulinho) [inline src/Core/Checkout/DependencyInjection/cart.xml]:
```suggestion
            <argument>%shopware.cart.expire_days%</argument>
```

File: src/Core/Checkout/DependencyInjection/cart.xml [hunk 3]
```diff
@@ -404,6 +405,7 @@
             <argument type="service" id="shopware.cart.redis" on-invalid="null"/>
             <argument type="service" id="Doctrine\DBAL\Connection"/>
             <argument>%shopware.cart.compress%</argument>
+            <argument>%shopware.cart.expire_days%%</argument>
```
Comment (keulinho) [inline src/Core/Checkout/DependencyInjection/cart.xml]:
```suggestion
            <argument>%shopware.cart.expire_days%</argument>
```

File: src/Core/Checkout/Cart/Command/CartMigrateCommand.php (hunk 1 above)
Comment (keulinho) [reply src/Core/Checkout/Cart/Command/CartMigrateCommand.php:239]: nope then it is fine as it is

Comment (Isengo1989) [reply src/Core/Checkout/Cart/Command/CartMigrateCommand.php:239]: @keulinho I change the cart.xml as suggested and added the expireDays to the tests. 

@shyim Questions on the "Missing tests" label. What type of test is missing here or was it referring to the missing 5th param?

File: tests/unit/php/Core/Checkout/Cart/Order/RedisCartPersisterTest.php [hunk 4]
```diff
@@ -27,7 +27,7 @@ public function testDecorated(): void
         $redis = $this->createMock(\Redis::class);
         $eventDispatcher = $this->createMock(EventDispatcher::class);
         $cartSerializationCleaner = $this->createMock(CartSerializationCleaner::class);
-        $persister = new RedisCartPersister($redis, $eventDispatcher, $cartSerializationCleaner, true);
+        $persister = new RedisCartPersister($redis, $eventDispatcher, $cartSerializationCleaner, true,90);
```
Comment (keulinho) [inline tests/unit/php/Core/Checkout/Cart/Order/RedisCartPersisterTest.php]:
```suggestion
        $persister = new RedisCartPersister($redis, $eventDispatcher, $cartSerializationCleaner, true, 90);
```
Code style, please fix the whole file or execute `ecs-fix`

File: src/Core/Checkout/Cart/Command/CartMigrateCommand.php (hunk 1 above)
Comment (keulinho) [reply src/Core/Checkout/Cart/Command/CartMigrateCommand.php:239]: you could adjust the tests in a way that you assert that redis is called with the correct `EX` value e.g.
```php
        $redis->expects(static::once())
            ->method('set')
            ->with(RedisCartPersister::PREFIX . $token, $content, YourEXValue);
 ```

**Pull request closed without merging**